app/src/generate.py

```diff
@@ -18,16 +20,29 @@
 - Keep your answers brief, a maximum of 5 sentences.
 - Keep your answers as similar to your knowledge text as you can.
 
+If the original question is in a language other than English, please provide your answer in the language of the original question.
+
 When referencing the context, do not quote directly.
 Use the provided citation numbers (e.g., (citation-1)) to indicate when you are drawing from the context.
-To cite multiple sources at once, you can append citations like so: (citation-1)(citation-2), etc.
-Place the citations after any closing punctuation for the sentence.
-For example: 'This is a sentence that draws on information from the context. (citation-1)'
+To cite multiple sources at once, you can append citations like so: (citation-1) (citation-2), etc.
+Place the citations immediately AFTER any closing punctuation for the sentence.
+For example: 'This is a sentence that draws on information from the context.(citation-1)'
+Do NOT place the citations BEFORE the closing punctuation, or add a space between the sentence and the citation.
 
 Example Answer:
 If the client lost their job at no fault, they may be eligible for unemployment insurance benefits. For example:
-- They may qualify if they were laid off due to lack of work. (citation-1)
-- They might be eligible if their hours were significantly reduced. (citation-2)
+- They may qualify if they were laid off due to lack of work.(citation-1) (citation-2)
+- They might be eligible if their hours were significantly reduced.(citation-3)
+"""
+
+ANALYZE_MESSAGE_PROMPT = """
+Analyze the user's message to determine how to respond.
+Reply with a JSON dictionary.
+Set original_language to the language of the user's message.
+If the user's message is in English, set is_in_english to true.
+Otherwise, set is_in_english to false and set message_in_english with the translation of the query into English.
+If the question would be easier to answer with additional policy or program context (such as policy documentation), set needs_context bool to True.
```

Comment:
The `bool` type isn't mentioned earlier in the prompt so let's keep it out for consistency unless you need it.
```suggestion
If the question would be easier to answer with additional policy or program context (such as policy documentation), set needs_context to True.
```